### DISCUSSION TITLE
save ocr in bubbles

### DISCUSSION THREAD

**Discussion by Shummuy** [VoxelCubes/PanelCleaner#4] [Q&A] [answered]
I've been using the program to clean up manga and then put a new translation on top, however, I already have this translation in other manga. I would like to know if there is any command to save the writings before making masks on top, it would greatly speed up the process I am carrying out. I know there's a way to save it in a .txt, however, I wanted to save them as images (keeping the page layout) to superimpose on the manga.

context: I'm copying a translation of a Naruto manga and putting it in a full color edition.

**Comment by VoxelCubes** [comment, marked as answer]
Awesome, glad you're enjoying it!

Oh, that's a great idea! PCleaner picks out the text, so it generates masks, and I could definitely use those masks to extract the text, rather than draw over it. I'll add that feature in the coming month, maybe even next week.

I can fit that in as an optional output between cleaning and denoising.

You could achieve the same effect manually by outputting the masks without denoising (using the -D flag), and then selecting the mask layer by alpha, then inverting the selection to delete everything except the text from the original image. Definitely annoying though, so I'll be sure to provide an automatic option soon.

**Reply by Shummuy** [reply]
That's good to hear, I tried that way you said with the -D but I couldn't do it. I hope it's not difficult to add an option like this, I believe it would optimize time in cases like mine (overlapping what's already done on something clean), at the moment I'm taking 25 minutes to redo chapters... (it was 35 before use pcleaner to clean the bubbles hehe)

**Reply by VoxelCubes** [reply]
Please check out version 1.2.0 where I added the -e or -extract-text flag to the cleaning process. This will do as described above. Let me know if this suits your needs.

**Reply by Shummuy** [reply]
It was wonderful! It turned out exactly as I imagined (even better, actually), and in initial tests, I was able to reduce the time it took to make each page by 50%. I couldn't find a configuration that would save only the text files (ignoring the clean and mask), but I was able to quickly solve it with a Ruby script. I am very grateful for the work, I searched every possible corner for something like this, and I can say that this is the best program for the purpose of cleaning (and now also isolating dialogues).

**Reply by VoxelCubes** [reply]
Awesome, I'm glad to hear that! I could definitely add another flag to only export the extracted text, not a problem. I'll let you know when that's done, probably in a day or two.

**Reply by VoxelCubes** [reply]
All right, give version 1.3.0 a try to use the -t or --save-only-text flag (it automatically sets the --extract-text flag as well) so that you really only get the text cut out. This update also includes a fix for file name clobbering, during the denoise step, such that files with the same name, but from different subdirectories can peacefully coexist in the working cache.

**Reply by Shummuy** [reply]
Awesome! Not having to run the second script will save even more time. I noticed there was a problem when running the script in more than one folder at the same time, it wouldn't work and I had to run them one by one. I didn't mention it before because I thought it might be my mistake in using the program, but the update fixed it. With the program in its current form, I reduced the time it takes to transfer manga texts from 35 minutes to 5 minutes, a huge gain in optimization. I'm really grateful for you making this program available and providing so much support for it!
